File: monWed_sectionA/week12_nov10-14/wedNov12/Person.java
public class Person implements Comparable<Person> {
    private String name;
    private int age;

    public Person(String n, int a) {
        name = n;
        age = a;
    }

    public String toString() {
        return String.format("Name: %s, age: %d%n", name, age);
    }

    public int compareTo(Person p) {
        if (name.compareTo(p.name) < 0) return -1;
        else if (name.compareTo(p.name) == 0) {
            if (age < p.age) return -1;
            else if (age == p.age) return 0;
            else return 1;
        }
        else {
            //name.compareTo(p.name) > 0
            return 1;
        }
    }
}

File: tuesThurs_sectionB/week14_dec1-5/tuesDec2/examples/Person.cs
public class Person {
    private int _age;

    //if no constructor, then get default for free:
    //public Person() {}

    public void Print() {
        Console.WriteLine($"Name: {Name}, age: {Age}");
    }

    //same thing for age, but only allow age 0-130
    public int Age {
        //get access and/or set access
        get {
            return _age;
        }
        set {
            //"value" means the information I am setting it to
            if (value >= 0 && value <= 130) {
                _age = value;
            }
        }
    }


    public string Name {get; set;}
}